hooks/checkout

```diff
@@ -69,22 +69,27 @@ checkout() {
   git reset --hard
   git clean -ffxdq
 
-  timeout "${GIT_REMOTE_TIMEOUT}" git fetch -v origin HEAD || exit_code=$?
-  check_timeout_exit_code "${exit_code}" "${GIT_REMOTE_TIMEOUT_EXIT_CODE}"
   git config remote.origin.fetch
-  timeout "${GIT_REMOTE_TIMEOUT}" git fetch -v --prune origin "${BUILDKITE_BRANCH}" || exit_code=$?
-  check_timeout_exit_code "${exit_code}" "${GIT_REMOTE_TIMEOUT_EXIT_CODE}"
 
   if [[ -z "${BUILDKITE_COMMIT:-}" || "${BUILDKITE_COMMIT}" == "HEAD" ]]; then
+    timeout "${GIT_REMOTE_TIMEOUT}" git fetch -v origin "${BUILDKITE_BRANCH}" || exit_code=$?
+    check_timeout_exit_code "${exit_code}" "${GIT_REMOTE_TIMEOUT_EXIT_CODE}"
     git checkout -f FETCH_HEAD
   else
-    # If the branch doesn't have the commit we want it might have been force
-    # pushed in the meantime. Exit with ESTALE to signify the stale branch
-    # reference in that case.
+    # If the commit isn't reachable the ref that was pointint to it might have
+    # been force pushed in the meantime. Exit with ESTALE to signify the stale
+    # branch reference in that case.
+    timeout "${GIT_REMOTE_TIMEOUT}" git fetch -v origin "${BUILDKITE_BRANCH}" || exit_code=$?
```

Comment:
I don't know how that happened but I ended up missing the one change this PR was supposed to do here. Will put up a new one.